reportportal_client/service.py

```diff
@@ -15,6 +15,8 @@
 """
 
 import json
+from time import sleep
```

Comment:
I'd use the [waiting library](https://pypi.org/project/waiting/) instead of implementing it.

Reply:
It is additional requirement to the project. I have my own implementations of diferent polling functions but it looks like overengineering. Do we really need it to cover one if?

Reply:
It does not cover only one "if" statement. It will replace the whole "for" cycle in the get_launch_info(). An example of "overengineering" is when you use ORM for doing one simple query.

Reply:
Hi,

Thanks for the comment. 
I still don`t think that adding one more dependency to the project to replace a few lines of trivial code, especially [waiting library](https://pypi.org/project/waiting/) worth it.

"waiting library";
1. does it has a gitlab with pipeline? (No)
2. Can we trust that code, is  there any big project that uses it? (I did not find any)
3. Where to file/fix bug? (I did not find any repo)
4. Who/how will support/test it for python 3.8+? 
5. We can`t control a logging messages or raising exceptions inside the waiting library.

```
Metadata-Version: 1.0
Name: waiting
Version: 1.4.1
Summary: Utility for waiting for stuff to happen
Home-page: UNKNOWN
Author: Rotem ***
Author-email: ***@gmail.com
License: BSD
Description: UNKNOWN
Platform: UNKNOWN
Classifier: Programming Language :: Python :: 2.6
Classifier: Programming Language :: Python :: 2.7
Classifier: Programming Language :: Python :: 3.3
Classifier: Programming Language :: Python :: 3.4
Classifier: Programming Language :: Python :: 3.5
```
And since you don`t want to rais an exception on getting information, we will have to wrap in in try...except and it will be about the same number of lines.

If you want to use the waiting library, could you please merge this request and add the waiting library later?

Reply:
I'm giving up...